torch/ao/quantization/fx/convert.py

```diff
@@ -491,27 +510,30 @@ def convert_custom_module(
     qconfig = observed_custom_module.qconfig
     if activation_is_statically_quantized(qconfig):
         statically_quantized_custom_module_nodes.add(node)
-        # remove the previous dequant node
-        prev_node = node.args[0]
-        # expecting the input node for a custom module node to be a Node
-        assert isinstance(prev_node, Node), \
-            f"Expecting the argument for custom module node to be a Node, but got {prev_node}"
-        if prev_node.op == "call_method" and prev_node.target == "dequantize":
-            # change the connection for custom module, we'll change the input
-            # of custom module node to quantize node:
-            # Before: quantize - dequantize - custom - module
-            # After: quantize - custom - module
-            #              \ - dequantize
-            node.replace_input_with(prev_node, prev_node.args[0])
-
-            # Remove the dequantize node if it doesn't have other users
-            if len(prev_node.users) == 0:
-                graph.erase_node(prev_node)
-
-        # absorb the following observer into the module conversion
-        activation_post_process = maybe_get_observer_for_node(node, modules)
-        assert activation_post_process is not None
-        observed_custom_module.activation_post_process = activation_post_process
+        if _is_custom_module_lstm(node, modules):
+            # The inputs are tuples in the form (input, (hidden0, hidden1))
+            # Ensure all three input nodes are quantized
+            assert (
+                len(node.args) == 2 and
+                isinstance(node.args[1], tuple) and
+                len(node.args[1]) == 2
+            )
+            (inputs, (hidden0, hidden1)) = node.args  # type: ignore[misc]
+            assert isinstance(inputs, Node)
+            assert isinstance(hidden0, Node)
+            assert isinstance(hidden1, Node)
+            _remove_previous_dequantize_in_custom_module(node, inputs, graph)
+            _remove_previous_dequantize_in_custom_module(node, hidden0, graph)
+            _remove_previous_dequantize_in_custom_module(node, hidden1, graph)
+        else:
+            # remove the previous dequant node to ensure the inputs are quantized
+            arg = node.args[0]
+            assert isinstance(arg, Node)
+            _remove_previous_dequantize_in_custom_module(node, arg, graph)
```

Comment:
maybe as a followup, we could also remove these code and instead in the prepare code, we add a QuantStub for the inputs that is consumed by lstm nodes, and in convert pass, we can just convert the QuantStub to quantize_per_tensor, this would be more aligned with long term plan I think

Reply:
It would also be good to value consistency with the current codebase, this would mean making this PR follow what the current codebase is doing (which is inserting observers and converting those to quant/dequant), and not include the "add qstub/dqstub" change into this PR.

Since both this change and the "add qstub/dqstub to FX graph mode quantization" are large, risky, and unrelated, it would be good to keep them separate.  Both of them run some risk of taking more time than anticipated, and if that happens it would be good to have less half-finished branches in the codebase.

Reply:
I agree consistency is important. However, using DeQuantStub here actually also significantly simplifies the matching logic in convert, otherwise we would have to do some additional matching to find the "special" observers, which adds some complexity. If you feel strongly about this then I can make the change, otherwise I would prefer to leave this.

Reply:
If we are going to break consistency, we should clearly document why we are doing this, and whether we expect the break in consistency to go away in the future or if it needs to stay around forever.  Can we add this documentation to both `prepare` and `convert` code?

Reply:
Sounds good, will do